app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModel.kt

```diff
@@ -465,8 +465,12 @@ class ZimManageViewModel @Inject constructor(
       .subscribe(
         { newList ->
           fileSelectListStates.postValue(
-            fileSelectListStates.value?.let { inheritSelections(it, newList) }
-              ?: FileSelectListState(newList)
+            fileSelectListStates.value?.let {
+              inheritSelections(
+                it,
+                newList as MutableList<BooksOnDiskListItem>
+              )
+            } ?: FileSelectListState(newList)
```

Comment:
Not sure, why do you want to implement this in this PR?

Reply:
there was an error after changing the code.
```
 Type mismatch: inferred type is List<BooksOnDiskListItem>! but MutableList<BooksOnDiskListItem> was expected
```